Can't build on Ubuntu 18.10 : Cannot read property 'toString' of null

Hello! I tried to build it and it failed.

I ran exactly these commands : 

```bash
git clone https://github.com/ipfs-shipyard/ipfs-desktop.git
cd ipfs-desktop
npm install
npm start
```

Here's the errors : 

```log
npm WARN acorn-jsx@5.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN eslint-config-aegir@1.0.1 requires a peer of eslint@^4.1.1 but none is installed. You must install peer dependencies yourself.
npm WARN You are using a pre-release version of node and things may not work as expected
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: macos-alias@0.2.11 (node_modules/macos-alias):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for macos-alias@0.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: appdmg@0.4.5 (node_modules/appdmg):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for appdmg@0.4.5: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 35176 packages in 13.15s
found 13 vulnerabilities (6 low, 6 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

```log
> ipfs-desktop@0.5.0 start /home/nato/Documents/GitHub/ipfs-desktop
> electron-forge start

✖ Checking your system

An unhandled error has occurred inside Forge:
Cannot read property 'toString' of null
TypeError: Cannot read property 'toString' of null
    at warnIfPackageManagerIsntAKnownGoodVersion (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-forge/dist/util/check-system.js:98:33)
    at /home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-forge/dist/util/check-system.js:44:9
    at process.internalTickCallback (internal/process/next_tick.js:77:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ipfs-desktop@0.5.0 start: `electron-forge start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ipfs-desktop@0.5.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nato/.npm/_logs/2018-11-01T02_40_15_938Z-debug.log
```

System info : 

```yaml
OS: Ubuntu 18.10 x86_64 
Kernel: 4.18.0-10-generic 
Uptime: 4 hours, 19 mins 
Packages: 2076 (dpkg), 41 (snap) 
Shell: bash 4.4.19 
Resolution: 1920x1080 
DE: GNOME 3.30.1 
WM: GNOME Shell 
WM Theme: Adwaita 
Theme: Adwaita-dark [GTK2/3] 
Icons: Yaru [GTK2/3] 
Terminal: gnome-terminal 
CPU: Intel i5-3570K (4) @ 4.100GHz 
GPU: NVIDIA GeForce GTX 660 Ti 
Memory: 4721MiB / 7921MiB
```

NPM : 

```json
{
	"npm": "6.4.1",
	"ares": "1.14.0",
	"cldr": "34.0",
	"http_parser": "2.8.0",
	"icu": "63.1",
	"modules": "67",
	"napi": "3",
	"nghttp2": "1.34.0",
	"node": "12.0.0-nightly20181031d690a87b8c",
	"openssl": "1.1.0i",
	"tz": "2018e",
	"unicode": "11.0",
	"uv": "1.23.2",
	"v8": "7.0.276.32-node.7",
	"zlib": "1.2.11"
}
```

Okay, here's what happens.

In order to be able to build the application, I need *first* to `npm i` on the master branch. Then, I need to switch to new and `npm i` again. There I'll be able to `npm start`. But there's plenty of errors and it doesn't look like what I expected the app to look like.

Also, before I get "You're using a preview of Node.js!", know that it does the exact same thing in both Node.js 11 and Node.js 12. I haven't tested legacy versions, however.

```log
npm i; npm start;
npm WARN npm npm does not support Node.js v12.0.0-nightly20181031d690a87b8c
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/1242/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> electron@3.0.7 postinstall /home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron
> node install.js


> ipfs-desktop@1.0.0 postinstall /home/nato/Documents/GitHub/ipfs-desktop
> electron-builder install-app-deps

  • electron-builder version=20.28.4
  • loaded configuration file=package.json ("build" field)
  • rebuilding native production dependencies platform=linux arch=x64
Error: /snap/node/1242/bin/node exited with code 7
Output:

> secp256k1@3.5.2 install /home/nato/Documents/GitHub/ipfs-desktop/node_modules/secp256k1
> npm run rebuild || echo "Secp256k1 bindings compilation fail. Pure JS implementation will be used."


> leveldown@3.0.2 install /home/nato/Documents/GitHub/ipfs-desktop/node_modules/leveldown
> prebuild-install || node-gyp rebuild


Error output:
npm WARN npm npm does not support Node.js v12.0.0-nightly20181031d690a87b8c
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/1242/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
npm ERR! code ELIFECYCLE
npm ERR! errno 7
npm ERR! leveldown@3.0.2 install: `prebuild-install || node-gyp rebuild`
npm ERR! Exit status 7
npm ERR!
npm ERR! Failed at the leveldown@3.0.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nato/.npm/_logs/2018-11-03T04_15_29_385Z-debug.log

    at ChildProcess.childProcess.once.code (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/builder-util/src/util.ts:254:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:969:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:257:5)
From previous event:
    at rebuild (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/app-builder-lib/out/util/yarn.js:239:18)
    at /home/nato/Documents/GitHub/ipfs-desktop/node_modules/app-builder-lib/src/util/yarn.ts:20:11
From previous event:
    at installOrRebuild (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/app-builder-lib/out/util/yarn.js:68:17)
    at /home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-builder/src/cli/install-app-deps.ts:56:9
    at Generator.next (<anonymous>)
From previous event:
    at installAppDeps (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-builder/out/cli/install-app-deps.js:174:17)
    at then (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-builder/src/cli/cli.ts:42:48)
    at processImmediate (timers.js:632:19)
From previous event:
    at Object.args [as handler] (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-builder/src/cli/cli.ts:42:48)
    at Object.runCommand (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/yargs/lib/command.js:238:44)
    at Object.parseArgs [as _parseArgs] (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/yargs/yargs.js:1059:30)
    at Object.get [as argv] (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/yargs/yargs.js:1000:21)
    at Object.<anonymous> (/home/nato/Documents/GitHub/ipfs-desktop/node_modules/electron-builder/src/cli/cli.ts:25:28)
    at Module._compile (internal/modules/cjs/loader.js:707:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:718:10)
    at Module.load (internal/modules/cjs/loader.js:605:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:544:12)
    at Function.Module._load (internal/modules/cjs/loader.js:536:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:760:12)
    at startup (internal/bootstrap/node.js:308:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:878:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ipfs-desktop@1.0.0 postinstall: `electron-builder install-app-deps`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ipfs-desktop@1.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nato/.npm/_logs/2018-11-03T04_15_29_449Z-debug.log
npm WARN npm npm does not support Node.js v12.0.0-nightly20181031d690a87b8c
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/
npm WARN lifecycle The node binary used for scripts is /snap/bin/node but npm is using /snap/node/1242/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> ipfs-desktop@1.0.0 start /home/nato/Documents/GitHub/ipfs-desktop
> cross-env NODE_ENV=development electron -r @babel/register src/index.js

2018-11-03T04:15:31.504Z info: Logger set up. Logs can be found on /home/nato/.config/Electron
2018-11-03T04:15:32.547Z info: Menubar is ready
2018-11-03T04:15:32.550Z info: Hash download shortcut enabled
2018-11-03T04:15:32.552Z info: Screenshot shortcut enabled
2018-11-03T04:15:32.565Z info: Launch on startup enabled

(electron:13684): libappindicator-WARNING **: 00:15:32.668: Using '/tmp' paths in SNAP environment will lead to unreadable resources
2018-11-03T04:15:37.622Z info: Disconnecting all IPFS instances
2018-11-03T04:15:37.623Z info: Done. Quitting app
```

--- a/src/utils/daemon.js
+++ b/src/utils/daemon.js
@@ -56,7 +56,13 @@ export default async function createDaemon (opts) {
     })
   }
 
-  let origins = await ipfsd.api.config.get('API.HTTPHeaders.Access-Control-Allow-Origin') || []
+  let origins = []
+  try {
+    origins = await ipfsd.api.config.get('API.HTTPHeaders.Access-Control-Allow-Origin')
+  } catch (e) {
+    logger.warn(e)
+  }
+
   if (!origins.includes('webui://-')) origins.push('webui://-')
   if (!origins.includes('https://webui.ipfs.io')) origins.push('https://webui.ipfs.io')
 

